### PR TITLE
Cleanup sim mergeable products

### PR DESCRIPTION
**Pull request by Dr15Jones**
Made non-critical changes to Sim Run products in order to clean them up. Removing the non-useful isMergeable method of GenRunInfoProduct will allow this product to be read at beginRun time and avoids the framework difficulties with mergeable run products.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by covarell** [comment]
@bendavid @syuvivida do you see any problem with this?

**Comment by bendavid** [comment]
This looks ok to me (but doesn't solve the issue for LHERunInfoProduct where it is both more critical and also more difficult to fix in a way which maintains compatibility both with existing samples and with the current wmLHE workflow)

**Comment by syuvivida** [comment]
@covarell I don't have problem with this. GenLumiInfo is not dependent on GenRunInfo.

File: SimDataFormats/GeneratorProducts/src/GenRunInfoProduct.cc [hunk 1]
```diff
@@ -21,34 +21,17 @@ GenRunInfoProduct::GenRunInfoProduct(GenRunInfoProduct const &other) :
...
 bool GenRunInfoProduct::isProductEqual(GenRunInfoProduct const &other) const
 {
-	return externalXSecLO_ == other.externalXSecLO_ &&
-	       externalXSecNLO_ == other.externalXSecNLO_ &&
-	       externalFilterEfficiency_ == other.externalFilterEfficiency_;
+	bool result =  externalXSecLO_ == other.externalXSecLO_ &&
+                       externalXSecNLO_ == other.externalXSecNLO_ &&
+	               externalFilterEfficiency_ == other.externalFilterEfficiency_;
+        if( not result) {
+          edm::LogWarning("GenRunInfoProduct|ProductsNotMergeable")
```
Comment (davidlange6) [inline SimDataFormats/GeneratorProducts/src/GenRunInfoProduct.cc]: @Dr15Jones - why not throw on this condition?

Comment (Dr15Jones) [inline SimDataFormats/GeneratorProducts/src/GenRunInfoProduct.cc]: As discussed in the Core meeting, the framework is not expecting this routine to throw and will handle the error when this function returns `false`.

**Pull request merged**